### PR TITLE
Rework hel headers

### PR DESCRIPTION
**Pull request by no92**

### CODE REVIEW

**Review by ArsenArsen — APPROVED** [review]
seems okay, unsure if those types were truly unused (but I can't find any reference to them), so someone else will have to fill in their purpose

File: hel/include/hel-types.h [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+//! Integer handle that represents a kernel resource.
+typedef int64_t HelHandle;
+
+typedef int64_t HelNanotime;
+
+
+enum {
+	// largest system call number plus 1
+	kHelNumCalls = 103,
```
Comment (avdgrinten) [inline hel/include/hel-types.h:22]: Syscall numbers are not really a "type" (so this should not be in `hel-types.h`).

File: hel/include/hel.h [hunk 2]
```diff
@@ -225,15 +130,6 @@ struct HelAction {
...
 };
 
-enum {
-	kHelDescMemory = 1,
-	kHelDescAddressSpace = 2,
-	kHelDescThread = 3,
-	kHelDescEndpoint = 5,
-	kHelDescIrq = 9,
-	kHelDescIo = 10,
-};
```
Comment (avdgrinten) [inline hel/include/hel.h:235]: I think these were returned by `helDescriptorInfo` in a previous iteration.

**Comment by no92** [comment]
Addressed the comments raised above.

**Review by Dennisbonke — APPROVED** [review]
LGTM

**Pull request merged**